### PR TITLE
Fix secretsmanager secret rotation failing because RotateImmediately is unset

### PR DESCRIPTION
**Pull request by cowlingj**
## Motivation

Issue: https://github.com/localstack/localstack/issues/9986

Secret rotation currently doesn't work when the `RotateImmediately` parameter is unset. It looks like this is because `backend_rotate_secret` receives the argument `rotate_immediately` with the value `None` when the parameter `RotateImmediately` is unset (if the argument is unset it defaults to `True`, but in this case it's being explicitly set to `None`), this causes the `new_version_id` to not be set resulting in the error described in the issue `cannot access local variable 'new_version_id' where it is not associated with a value`.

## Changes

Check if rotate_immediately is `None` and set it to `True` in that case

## Testing

I'm unable to run any of the tests locally, so that'll be worth paying extra attention to when reviewing.

### CODE REVIEW

**Comment by cowlingj** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by dominikschubert** [comment]
Rebased on master & generated updated snapshots for the parametrized test.

File: tests/aws/services/secretsmanager/test_secretsmanager.py [hunk 1]
```diff
@@ -367,13 +378,17 @@ def test_rotate_secret_with_lambda_success(
...
+            rotation_kwargs["RotateImmediately"] = rotate_immediately
         rot_res = aws_client.secretsmanager.rotate_secret(
             SecretId=secret_name,
             RotationLambdaARN=function_arn,
             RotationRules={
                 "AutomaticallyAfterDays": 1,
             },
-            RotateImmediately=True,
+            **rotation_kwargs
+            # RotateImmediately=False,
```
Comment (dominikschubert) [inline tests/aws/services/secretsmanager/test_secretsmanager.py:391]:
```suggestion
```
small leftover from my test edit

**Review by dominikschubert — APPROVED** [review]
LGTM, thanks for fixing the default behavior for the RotateImmediately field!

There's a few failing tests right now but those are caused by  an opensearch flake.

**Pull request merged**